src/Components/Artwork/Save.tsx

```diff
@@ -1,92 +1,154 @@
   useRelay?: boolean
 }
 
-export class SaveButtonContainer extends React.Component<Props, null> {
-  static defaultProps = {
-    useRelay: true,
-  }
+interface State {
+  is_saved: boolean
+}
+
+export const SaveButtonContainer = Artsy.ContextConsumer(
+  class extends React.Component<Props, State> {
+    static defaultProps = {
+      useRelay: true,
+    }
+
+    state = {
+      is_saved: null,
+    }
+
+    get isSaved() {
+      const isSaved = isNull(this.state.is_saved)
+        ? this.props.artwork.is_saved
+        : this.state.is_saved
+
+      return isSaved
+    }
+
+    handleSave() {
+      const {
+        currentUser,
+        artwork,
+        relay,
+        relayEnvironment,
+        useRelay,
+      } = this.props
+      const environment = (relay && relay.environment) || relayEnvironment
 
-  handleSave() {
-    const { currentUser, artwork, relay, useRelay } = this.props
-    if (useRelay && currentUser && currentUser.id) {
-      commitMutation(relay.environment, {
-        mutation: graphql`
-          mutation SaveArtworkMutation($input: SaveArtworkInput!) {
-            saveArtwork(input: $input) {
-              artwork {
-                is_saved
+      if (environment && currentUser && currentUser.id) {
+        // Optimistic update for environments that don't have typical access to
+        // Relay, e.g., where new ArtworkGrids are used in old code via Stitch. Note
+        // that the prop `useRelay` refers to outer HOC wrappers. In cases where
+        // Save UI components are used it is possible to piggyback on ContextProvider
+        // environment for mutations, but since the component exists outside of a
+        // Relay HOC props are not updated when successful mutations occur -- hence
+        // the need for setState.
+        //
+        // TODO:
+        // Refactor out `useRelay` prop when Force artwork Grids have been moved
+        // completely over to Relay
+
+        if (!useRelay) {
+          this.setState({
+            is_saved: !this.isSaved,
+          })
+        }
+
+        commitMutation(environment, {
+          mutation: graphql`
+            mutation SaveArtworkMutation($input: SaveArtworkInput!) {
+              saveArtwork(input: $input) {
+                artwork {
+                  id
+                  is_saved
+                }
               }
             }
-          }
-        `,
-        variables: {
-          input: {
-            artwork_id: artwork.id,
-            remove: this.props.artwork.is_saved,
+          `,
+          variables: {
+            input: {
+              artwork_id: artwork.id,
+              remove: this.isSaved,
+            },
           },
-        },
-        // TODO: Relay Modern: This is not working yet
-        optimisticResponse: {
-          saveArtwork: {
-            artwork: {
-              __id: artwork.__id,
-              is_saved: !this.props.artwork.is_saved,
+          // TODO: Relay Modern: This is not working yet
```

Comment:
This actually does work, no? So this comment seems out of date?

Reply:
Will remove